prototype2/efu/HwCheck.h

```diff
@@ -0,0 +1,46 @@
+/** Copyright (C) 2018 European Spallation Source */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// Class used to perform OS and Hardware related checks prior to
+/// application start
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <net/if.h>
+
+class HwCheck {
+public:
```

Comment:
In my opinion, having this be a C-function is better than having it be a C++ class. The reason being that there is very little state that is kept track of and the member function is only called once.

Reply:
I expect this class to evolve soon to contain more hardware related checks and hence more member functions.

Reply:
The question is still: do you expect the member functions to share state and do you expect that they will be called more than once per application run?